### ISSUE TITLE
Events are excluded when EXDATE == DTSTART

### ISSUE THREAD

**Issue by josh-philpott-gp** [u01jmg3/ics-parser#298]
Hi!

I'm encountering an issue where events are excluded when an EXDATE == DTSTART (see: https://github.com/u01jmg3/ics-parser/issues/240).

According to [RFC5545](https://datatracker.ietf.org/doc/html/rfc5545#section-3.8.5.1), this is a valid event:

> The "EXDATE" property can be used to exclude the value specified
      in "DTSTART".  However, in such cases, the original "DTSTART" date
      MUST still be maintained by the calendaring and scheduling system
      because the original "DTSTART" value has inherent usage
      dependencies by other properties such as the "RECURRENCE-ID".

I've encountered this scenario in Outlook & Apple calendar feeds when the first recurring event is deleted followed by cancelling the rest of the recurring events.

**Comment by s0600204** [comment]
Could you give us an example or excerpt of a calendar file that causes the `ics-parser` to exhibit the behaviour you describe?

Could you also tell us what versions of `php` and the `ics-parser` you are using?

**Comment by josh-philpott-gp** [comment]
`PHP Version`: 5.6.40
`ICS Parser Version`: dev-master (originally found in 2.1.19 - it looks like this behavior was introduced in 2.1.16)

Here's the relevant portion of the ics file. 

```METHOD:CANCEL
BEGIN:VEVENT
UID:B5F1A317-A9D9-40FB-9AB8-EFABFBCDFC01
DTEND;TZID=America/Los_Angeles:20211210T103000
SUMMARY:Test Reoccurring
LAST-MODIFIED:20211209T230242Z
CREATED:20211209T223201Z
ORGANIZER;CN=Kevin Lisbin;EMAIL=kevin.lisbin@gmail.com:mailto:2_GE3DEMBWG
 E3DIMJZG4YTMMRQGYHFIK5RTDQSLGCNN5WXBI3WSWS26CS3POVOKHVV3XKK6O4PUZHES@imi
 p.me.com
SEQUENCE:2
DTSTART;TZID=America/Los_Angeles:20211210T100000
RRULE:FREQ=MONTHLY
EXDATE;TZID=America/Los_Angeles:20211210T100000
ATTENDEE;EMAIL=kevin+fastGPA@grandpad.net;CUTYPE=INDIVIDUAL;PARTSTAT=NEED
 S-ACTION;SCHEDULE-STATUS=1.1:mailto:kevin+fastGPA@grandpad.net
DTSTAMP:20211209T231616Z
STATUS:CANCELLED
END:VEVENT
```

[Full .ics file](https://github.com/u01jmg3/ics-parser/files/8189211/Delete.remaining.reoccurrances.txt)

**Comment by s0600204** [comment]
The example you have given generates no events because the `ics-parser` does not currently support Recurrence Rules with Monthly frequency and no `BY{*}` modifiers. [Easily remedied.](https://github.com/u01jmg3/ics-parser/pull/301)

Are there any other examples you know of that exhibit the behaviour you described in the initial report?

**Comment by u01jmg3** [comment]
Relates to #220

**Comment by u01jmg3** [comment]
Closed by PR #301

**Comment by josh-philpott-gp** [comment]
No, that was it and seems to do the trick. Thank you for taking a look!
